frontend/vue/components/CodeExercise/CodeEditor.vue

```diff
@@ -0,0 +1,148 @@
+<template>
+  <div class="code-editor">
+    <CodeArea class="code-editor__text-area" :code="internalCode" @codeChanged="codeChanged" />
+    <CodeEditorTools
+      class="code-editor__tools"
+      :copy-text="internalCode"
+      :copy-enabled="copyEnabled"
+      :reset-enabled="resetEnabled"
+      :notebook-enabled="notebookEnabled"
+      @reset="resetRequest"
+      @notebook="notebook"
+    />
+    <div
+      v-if="resetNotificationOpen"
+      class="code-editor__reset-notification__wrapper"
+    >
+      <bx-toast-notification
+        class="code-editor__reset-notification"
+        title="Reset code block?"
```

Comment:
For these attributes, I think we could provide translations, something like:

```
title="$translate('Reset code block?')"
icon-description="$translate('Cancel')"
subtitle="$translate('This will reset the code to the textbook default. Any custom edits will be removed.')"
```

then make sure the strings are defined in `strings.yaml`

(also note I changed `de` to `the`)